### PR TITLE
Added PHP 8.0 support #4

### LINKED ISSUE #4: PHP 8.0 support
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes

#### Summary

To be prepared for the december release of PHP 8.0, this repository has some additional TODOs to be tested against the new major version.

In order to make this repository compatible, one has to follow these steps:

- [ ] Modify `composer.json` to provide support for PHP 8.0 by adding the constraint `~8.0.0`
- [ ] Modify `composer.json` to drop support for PHP less than 7.3
- [ ] Modify `composer.json` to implement phpunit 9.3 which supports PHP 7.3+
- [ ] Modify `.travis.yml` to ignore platform requirements when installing composer dependencies (simply add `--ignore-platform-reqs` to `COMPOSER_ARGS` env variable)
- [ ] Modify `.travis.yml` to add PHP 8.0 to the matrix (*NOTE*: Do not allow failures as PHP 8.0 has a feature freeze since 2020-08-04!)
- [ ] Modify source code in case there are incompatibilities with PHP 8.0

### PR DESCRIPTION
**Pull request by svycka**
Q | A
-- | --
New Feature | yes

Summary
To be prepared for the december release of PHP 8.0, this repository has some additional TODOs to be tested against the new major version.

In order to make this repository compatible, one has to follow these steps:

- [x] Modify composer.json to provide support for PHP 8.0 by adding the constraint ~8.0.0
- [x] Modify composer.json to drop support for PHP less than 7.3
- [x] Modify composer.json to implement phpunit 9.3 which supports PHP 7.3+
- [x] Modify .travis.yml to ignore platform requirements when installing composer dependencies (simply add --ignore-platform-reqs to COMPOSER_ARGS env variable)
- [x] Modify .travis.yml to add PHP 8.0 to the matrix (NOTE: Do not allow failures as PHP 8.0 has a feature freeze since 2020-08-04!)
- [x] Modify source code in case there are incompatibilities with PHP 8.0

closes #4

### CODE REVIEW

File: test/BodyParams/BodyParamsMiddlewareTest.php [hunk 1]
```diff
@@ -18,16 +18,20 @@
 use Mezzio\Helper\Exception\MalformedRequestBodyException;
 use PHPUnit\Framework\TestCase;
 use Prophecy\Argument;
+use Prophecy\PhpUnit\ProphecyTrait;
 use Psr\Http\Message\ServerRequestInterface;
 use Psr\Http\Server\RequestHandlerInterface;
 
+use ReflectionProperty;
```
Comment (samsonasik) [inline test/BodyParams/BodyParamsMiddlewareTest.php:24]: can be added after line 23 directly

File: .travis.yml [hunk 2]
```diff
@@ -6,38 +6,38 @@ cache:
 
 env:
   global:
-    - COMPOSER_ARGS="--no-interaction"
+    - COMPOSER_ARGS="--no-interaction --ignore-platform-reqs"
```
Comment (froschdesign) [inline .travis.yml]: The new parameter is only needed for the nightly version. Please move this to the related section.

File: .travis.yml [hunk 3]
```diff
@@ -6,38 +6,38 @@ cache:
...
       env:
         - DEPS=lowest
-    - php: 7.2
+    - php: 7.4
       env:
         - DEPS=latest
-    - php: 7.3
+    - php: nightly
       env:
         - DEPS=lowest
```
Comment (froschdesign) [inline .travis.yml:31]:
```suggestion
    - php: nightly
      env:
        - DEPS=lowest
        - COMPOSER_ARGS="--no-interaction --ignore-platform-reqs"
```

File: .travis.yml [hunk 4]
```diff
@@ -6,38 +6,38 @@ cache:
...
-    - php: 7.3
+    - php: nightly
       env:
         - DEPS=latest
```
Comment (froschdesign) [inline .travis.yml:35]:
```suggestion
    - php: nightly
      env:
        - DEPS=latest
        - COMPOSER_ARGS="--no-interaction --ignore-platform-reqs"
```

File: phpunit.xml.dist [hunk 5]
```diff
@@ -1,17 +1,13 @@
...
-            <directory>./test</directory>
-        </testsuite>
-    </testsuites>
-
-    <filter>
-        <whitelist processUncoveredFilesFromWhitelist="true">
-            <directory suffix=".php">src</directory>
-        </whitelist>
-    </filter>
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.3/phpunit.xsd" bootstrap="vendor/autoload.php" colors="true">
```
Comment (froschdesign) [inline phpunit.xml.dist]: The local XML schema definition can be used here:
```suggestion
<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd" bootstrap="vendor/autoload.php" colors="true">
```

**Review by froschdesign — CHANGES_REQUESTED** [review]
Thanks for your contribution! 👍 

This is a first review with some small hints.

**Review by boesing — APPROVED** [review]
LGTM

**Review by samsonasik — APPROVED** [review]

**Comment by boesing** [comment]
Thanks, @svycka!

**Pull request merged**